### PR TITLE
Updated Villager Mapping

### PR DESCRIPTION
**Pull request by Gegy**

### CODE REVIEW

File: mappings/net/minecraft/entity/player/PlayerEntity.mapping [hunk 1]
```diff
@@ -33,6 +33,7 @@ CLASS aun net/minecraft/entity/player/PlayerEntity
 	FIELD g spawnPosition Leu;
 	METHOD a sendVillagerRecipes (ILbfc;IIZ)V
 		ARG 1 syncId
+		ARG 5 canLevel
```
Comment (liach) [inline mappings/net/minecraft/entity/player/PlayerEntity.mapping]: Isn't this like showing level in screen which is true for villagers but false for wandering traders?

File: mappings/net/minecraft/util/math/GlobalPos.mapping [hunk 2]
```diff
@@ -0,0 +1,14 @@
+CLASS fb net/minecraft/util/math/GlobalPos
```
Comment (liach) [inline mappings/net/minecraft/util/math/GlobalPos.mapping]: `UniversalPos`? `Global` does not fit the minecraft theme.

File: mappings/net/minecraft/village/VillagerData.mapping [hunk 3]
```diff
@@ -1,11 +1,25 @@
...
 	METHOD a withType (Lauf;)Laub;
+		ARG 1 type
 	METHOD a serialize (Lcom/mojang/datafixers/types/DynamicOps;)Ljava/lang/Object;
 	METHOD b getProfession ()Laud;
+	METHOD b getLowerLevelExperience (I)I
+		ARG 0 level
 	METHOD c getLevel ()I
+	METHOD c getUpperLevelExperience (I)I
+		ARG 0 level
+	METHOD d isLevelValid (I)Z
```
Comment (liach) [inline mappings/net/minecraft/village/VillagerData.mapping:21]: I remember this one returns false for stone level. Is this `hasLevelledUp`?

**Review by liach — APPROVED** [review]

File: mappings/net/minecraft/entity/player/PlayerEntity.mapping (hunk 1 above)
Comment (Gegy) [reply mappings/net/minecraft/entity/player/PlayerEntity.mapping]: Yep

File: mappings/net/minecraft/util/math/GlobalPos.mapping (hunk 2 above)
Comment (Gegy) [reply mappings/net/minecraft/util/math/GlobalPos.mapping]: The Mojang name is `GlobalPos`, it seems.

File: mappings/net/minecraft/village/VillagerData.mapping (hunk 3 above)
Comment (Gegy) [reply mappings/net/minecraft/village/VillagerData.mapping:21]: Oh, potentially. Will have to check that, thanks!

File: mappings/net/minecraft/entity/ai/brain/EntityMemoryModule.mapping [hunk 4]
```diff
@@ -0,0 +1,10 @@
+CLASS anw net/minecraft/entity/ai/brain/EntityMemoryModule
```
Comment (Runemoro) [inline mappings/net/minecraft/entity/ai/brain/EntityMemoryModule.mapping]: The vanilla name is MemoryModule**Type** (based on name in `Registry`). It doesn't actually store any data, it's used as a key to get the memory module.

**Comment by Gegy** [comment]
This has pretty much been superseded by #533 -- once that is merged I'll make a PR to add the changes from here not common between the two.

**Comment by liach** [comment]
hmm? why reopen?

**Comment by Gegy** [comment]
Will merge into this PR

File: mappings/net/minecraft/village/VillagerData.mapping (hunk 3 above)
Comment (Gegy) [reply mappings/net/minecraft/village/VillagerData.mapping:21]: Seems the minimum level value is `1`

File: mappings/net/minecraft/entity/ai/brain/Brain.mapping [hunk 5]
```diff
@@ -20,3 +25,4 @@ CLASS aiu net/minecraft/entity/ai/brain/Brain
 	METHOD b stopAllTasks (Lvc;Laif;)V
 	METHOD c getMemory (Lanw;)Ljava/util/Optional;
 	METHOD c hasActivity (Lavw;)Z
+	METHOD e clone ()Laiu;
```
Comment (liach) [inline mappings/net/minecraft/entity/ai/brain/Brain.mapping:28]: Are you kidding me... This is probably the worst mapping that can ever happen.

Comment (Gegy) [reply mappings/net/minecraft/entity/ai/brain/Brain.mapping:28]: elaborate?

Comment (liach) [reply mappings/net/minecraft/entity/ai/brain/Brain.mapping:28]: Looks like it overrides `Object#clone` by accident. #546

Comment (grondag) [reply mappings/net/minecraft/entity/ai/brain/Brain.mapping:28]: You’re probably right, but curious: if Mojang did actually override clone() would it be obfuscated? Do we have a reliable way to know if this is or isn’t clone()?

Comment (Gegy) [reply mappings/net/minecraft/entity/ai/brain/Brain.mapping:28]: Oops. 😄 
Apologies about that..

Comment (liach) [reply mappings/net/minecraft/entity/ai/brain/Brain.mapping:28]: It shouldn't be. I think now enigma correctly handles synthetic methods generated for such overrides that changes the return type in the signature (see metadata section reader classes)

**Pull request merged**